File: 167-TwoSumIiInputArrayIsSorted/167-TwoSumIiInputArrayIsSorted.py
# Last updated: 19/2/2026, 9:14:10 am
class Solution:
    def twoSum(self, numbers: List[int], target: int) -> List[int]:
        r=0
        l=len(numbers)-1
        while(r<l):
            temp=numbers[r]+numbers[l]
            if(temp < target ):
                r+=1
            elif(temp >target):
                l-=1
            else:
                return([r+1,l+1])

File: 167-TwoSumII-InputArrayIsSorted/167-TwoSumII-InputArrayIsSorted.cpp
// Last updated: 28/7/2026, 12:48:06 pm
1class Solution {
2public:
3    vector<int> twoSum(vector<int>& numbers, int target) {
4        int low=0;
5        int high=numbers.size()-1;
6        while(low<high){
7            int sum=numbers[low]+numbers[high];
8            if(sum==target){
9                return{low+1,high+1};
10            }
11            else if(sum <target){
12                low++;
13            }
14            else{
15                high--;
16            }
17        }
18        return{-1,-1};
19        
20    }
21};